controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/SystemUpgrader.java

```diff
@@ -75,12 +73,16 @@ private void deploy(Version target, SystemApplication application, ZoneId zone)
         }
     }
 
+    private boolean convergedOn(ZoneId zone, SystemApplication application, Version target) {
+        return currentVersion(zone, application.id(), target).equals(target);
+    }
+
     private Version wantedVersion(ZoneId zone, ApplicationId application, Version defaultVersion) {
         return minVersion(zone, application, Node::wantedVersion).orElse(defaultVersion);
     }
 
-    private Version currentVersion(ZoneId zone, ApplicationId application) {
-        return minVersion(zone, application, Node::currentVersion).orElse(Version.emptyVersion);
+    private Version currentVersion(ZoneId zone, ApplicationId application, Version defaultVersion) {
+        return minVersion(zone, application, Node::currentVersion).orElse(defaultVersion);
```

Comment:
Exceptions are caught in the client, and result in `empty()`. This needs to change first, i.e., catch only when expected.

Reply:
Fixed, PTAL.